client/src/api/block_builder/input_selection/core/burn.rs

```diff
@@ -111,28 +111,48 @@ impl Burn {
 #[serde(rename_all = "camelCase")]
 pub struct BurnDto {
     /// Aliases to burn.
-    pub(crate) aliases: HashSet<AliasId>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) aliases: Option<HashSet<AliasId>>,
     /// NFTs to burn.
-    pub(crate) nfts: HashSet<NftId>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) nfts: Option<HashSet<NftId>>,
     /// Foundries to burn.
-    pub(crate) foundries: HashSet<FoundryId>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) foundries: Option<HashSet<FoundryId>>,
     /// Amounts of native tokens to burn.
     /// `hashbrown::HashMap` to allow seamless operations with `NativeTokens`.
-    pub(crate) native_tokens: HashMap<TokenId, U256Dto>,
+    #[serde(skip_serializing_if = "Option::is_none")]
+    pub(crate) native_tokens: Option<HashMap<TokenId, U256Dto>>,
 }
 
 impl From<&Burn> for BurnDto {
     fn from(value: &Burn) -> Self {
         Self {
-            aliases: value.aliases.clone(),
-            nfts: value.nfts.clone(),
-            foundries: value.foundries.clone(),
-            native_tokens: HashMap::from_iter(
-                value
-                    .native_tokens
-                    .iter()
-                    .map(|(token_id, amount)| (*token_id, U256Dto::from(amount))),
-            ),
+            aliases: if value.aliases.is_empty() {
+                None
+            } else {
+                Some(value.aliases.clone())
+            },
```

Comment:
Does this work and would you prefer that?
```suggestion
            aliases: value.aliases.is_empty().then_some(value.aliases.clone()),
```

Reply:
But that would do `Some` when it's empty

Reply:
Unfortunately there is no `is_full` or whatever

Reply:
Ah yeah, but what about `!value.aliases.is_empty()`?

Reply:
Should work, I didn't think of that

Reply:
Don't think I ever used then_some before, good stuff

Reply:
Done